webapp/views/App/views/Designer/TaxonomyDetail/store/useTaxonomyDetail.js

```diff
@@ -1,18 +1,19 @@
 import { useState, useEffect } from 'react'
-
+import { useDispatch } from 'react-redux'
 import { useActions } from './actions'
-import { State } from './state'
+import * as TaxonomyActions from '../actions'
 
 export const useTaxonomyDetail = (props) => {
-  const { taxonomy } = props
+  const { onTaxonomyCreate } = props
+  const dispatch = useDispatch()
+  const [state, setState] = useState(null)
```

Comment:
I would initialize with empty object. you will avoid that empty check in the component